reporter.go

```diff
@@ -199,7 +199,7 @@ func NewRemoteReporter(sender transport.Transport, options *ReporterOptions) Rep
 		options.Logger = NullLogger
 	}
 	if options.Metrics == nil {
```

Comment:
given that we're changing the API, does it make sense to convert reporter to functional options? (in a different pr)